src/main/java/io/vertx/ext/jdbc/impl/actions/JDBCSQLRowStream.java

```diff
@@ -103,18 +109,20 @@ public SQLRowStream resume() {
   }
 
   private void nextRow() {
+    while (!paused.get() && !accumulator.isEmpty()) {
```

Comment:
add a comment that says that paused.get act as volatile read / memory barrier and that it must be done before the accumulator read.

we should check also that `paused.get() && !accumulator.isEmpty()` cannot be reordered because I am not sure of how this happens in a A && B expression